src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
+enum Shapes {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle'
+}
+
+enum Colors {
+  red = 'red',
+  green = 'green',
+  blue = 'blue'
+}
+
 export interface Figure {
+  shape: Shapes;
+  color: Colors;
 
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shapes.triangle = Shapes.triangle;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const arayOfSides = [a, b, c]
+      .sort((number1: number, number2: number) => number2 - number1);
+    const invalidTriangle = arayOfSides[0] >= (arayOfSides[1] + arayOfSides[2]);
+
+    if (invalidTriangle === true) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
 
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('invalid side length');
+    }
```

Comment:
Better to do this check first, at the very beggining. Because if some side is < 0 -> we will not need to do sorting and other calculations